WalletWasabi/WabiSabi/Backend/Rounds/Arena.cs

```diff
@@ -326,26 +326,30 @@ public async Task<InputRegistrationResponse> RegisterInputAsync(InputRegistratio
 
 		public async Task ReadyToSignAsync(ReadyToSignRequestRequest request, CancellationToken cancellationToken)
 		{
+			Alice? alice;
+
 			using (await AsyncLock.LockAsync(cancellationToken).ConfigureAwait(false))
 			{
-				if (Rounds.FirstOrDefault(r => r.Id == request.RoundId) is not Round round)
+				var round = Rounds.FirstOrDefault(r => r.Id == request.RoundId);
+				if (round is null)
 				{
 					throw new WabiSabiProtocolException(WabiSabiProtocolErrorCode.RoundNotFound, $"Round ({request.RoundId}) not found.");
 				}
 
-				if (round.Alices.FirstOrDefault(a => a.Id == request.AliceId) is not Alice alice)
+				alice = round.Alices.FirstOrDefault(a => a.Id == request.AliceId);
+				if (alice is null)
 				{
 					throw new WabiSabiProtocolException(WabiSabiProtocolErrorCode.AliceNotFound, $"Round ({request.RoundId}): Alice id ({request.AliceId}).");
 				}
```

Comment:
Once `alice` is has signalled readyness, any additional verification of ownership proofs is wasteful.

Furthermore, because POST requests are idempotent, this can be interpreted as a sign of malice. We can even restore `Prison.Ban` (recently removed due to lack of use) and use it here instead of `Note`, and throw an error instead of returning.

```suggestion
				}
				
				if (alice.ReadyToSign)
				{
					Prison.Note(alice, round.Id);
					return;
				}
```

Reply:
https://github.com/zkSNACKs/WalletWasabi/pull/6161